stock_average_daily_sale/models/stock_average_daily_sale.py

```diff
@@ -309,13 +416,19 @@ def _create_materialized_view(self):
                         GREATEST(
                             (cfg.number_days_qty_in_stock * average_qty_by_sale * average_daily_sales_count) + (ds.daily_standard_deviation * cfg.safety_factor * sqrt(nbr_days)),
                             (cfg.number_days_qty_in_stock *  average_qty_by_sale)
-                        ) as recommended_qty
+                        ) as recommended_qty,
+                        GREATEST(
+                            (cfg.number_days_qty_in_stock * (average_qty_by_sale - COALESCE(average_qty_by_return, 0)) * (average_daily_sales_count - COALESCE(average_daily_returns_count, 0))) + ((ds.daily_standard_deviation - COALESCE(dsr.daily_standard_deviation, 0)) * cfg.safety_factor * sqrt(nbr_days)),
```

Comment:
This new formula looks wrong to me
If you always ship 10 and you return also 10, then `average_qty_by_sale - average_qty_by_return = 0`. So a single return will put the recommended qty to 0. That's wrong.
You need to compute the daily sale - the daily return
```suggestion
                            (cfg.number_days_qty_in_stock * (
    average_qty_by_sale * average_daily_sales_count
    - COALESCE(average_qty_by_return, 0) * COALESCE(average_daily_returns_count, 0)
                            )  + ((ds.daily_standard_deviation - COALESCE(dsr.daily_standard_deviation, 0)) * cfg.safety_factor * sqrt(nbr_days)),
```

Reply:
@sebalix is the author of these changes, but I also know the topic so I will clarify.
We have 'hidden' the returns handling because the way we handle returns is not universal here. In this case, as returns, we consider moves where `sl_src.usage in ('inventory') AND sl_dest.usage in ('internal')` which is rather a custom requirement.
The changes you requested are still relevant in my opinion though. Thank you for the suggestion, I will fix that.

Reply:
Thanks @twalter-c2c 
I'm still clarifying the formula with Meredith